packages/plugin-hardhat/src/utils/deploy-impl.ts

```diff
@@ -20,7 +19,7 @@ import { readValidations } from './validations';
 
 export interface DeployedImpl {
   impl: string;
-  txResponse?: ethers.providers.TransactionResponse;
+  txResponse?: null | ethers.TransactionResponse;
```

Comment:
It'd be better to avoid `null` here. The main issue I see is that if we have multiple representations for "not present", checking for `txResponse !== undefined` is not exhaustive. The type system should catch that, and I can see that you adapted to this in a few places, but it all seems unnecessary if we can just consolidate it into `undefined` in the functions below:

```diff
- txResponse = await hre.ethers.provider.getTransaction(deployment.txHash);
+ txResponse = await hre.ethers.provider.getTransaction(deployment.txHash) ?? undefined;
```